protos/core.proto

```diff
@@ -260,127 +250,422 @@ message CompiledGraph {
+
+    // The keys by which to cluster partitions by. See
+    // https://cloud.google.com/dataform/docs/partitions-clusters.
+    repeated string cluster_by = 2;
+
+    // SQL-based filter for when incremental updates are applied.
+    string update_partition_filter = 3;
+
+    // Key-value pairs for BigQuery labels. If the label name contains special
+    // characters, e.g. hyphens, then quote its name, e.g. `labels: {
+    // "label-name": "value" }`.
+    map<string, string> labels = 4;
+
+    // The number of days for which BigQuery stores data in each partition. The
+    // setting applies to all partitions in a table, but is calculated
+    // independently for each partition based on the partition time.
+    int32 partition_expiration_days = 5;
+
+    // Declares whether the partitioned table requires a WHERE clause predicate
+    // filter that filters the partitioning column.
+    bool require_partition_filter = 6;
+
+    // Key-value pairs for the table, view, and materialized view options.
+    map<string, string> additional_options = 7;
   }
-}
 
-message CompileExecutionRequest {
-  CompileConfig compile_config = 1;
-}
+  message Table {
+    // Target is the unique action identifier.
+    Target target = 1;
 
-message CompileExecutionResponse {
-  CompiledGraph compiled_graph = 1;
-}
+    // Canonical Target is the unique action identifier, with no target
+    // overrides applied.
+    // This option cannot be manually specified in action configs.
+    Target canonical_target = 2;
 
-// This feature list is added to when making potentilly backwards breaking
-// changes. It lets the caller of Dataform Core know whether it supports the
-// change.
-enum SupportedFeatures {
-  UNKNOWN_FEATURE = 0;
-  ARRAY_BUFFER_IPC = 1;
-}
+    // Targets (non-canonical) of actions that this action is dependent on.
+    repeated Target dependency_targets = 3;
 
-message ActionConfigs {
-  repeated ActionConfig actions = 1;
-}
+    // Path to the source file that the contents of the action is loaded from.
+    string source = 4;
 
-message ActionConfig {
-  // Target is a unique action identifier.
-  Target target = 1;
+    // Path that the config of the action is loaded from.
+    // This option cannot be manually specified in action configs.
+    string config_path = 5;
 
-  // A list of user-defined tags with which the action should be labeled.
-  repeated string tags = 2;
+    // A list of user-defined tags with which the action should be labeled.
+    repeated string tags = 6;
 
-  // Targets of actions that this action is dependent on.
-  repeated Target dependency_targets = 3;
+    // If set to true, this action will not be executed. However, the action may
+    // still be depended upon.
+    bool disabled = 7;
 
-  // Filename to load the action contents from.
-  string file_name = 4;
+    // Describes a struct, object or record in a dataset that has nested
+    // columns.
+    ActionDescriptor action_descriptor = 8;
```

Comment:
This contains description and column descriptions,  I don't think you need them below? or vice versa

Reply:
Ah yep! That was the whole point of me flattening the fields under the actions